sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/CommentKeywordBase.cs

```diff
@@ -0,0 +1,122 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2019 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Text;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.Rules
+{
+    public abstract class CommentKeywordBase : SonarDiagnosticAnalyzer
+    {
+        private const string TODO_Keyword = "TODO";
+        protected const string TODO_DiagnosticId = "S1135";
+        protected const string TODO_MessageFormat = "Complete the task associated to this '" + TODO_Keyword + "' comment.";
+
+        private const string FIXME_Keyword = "FIXME";
+        protected const string FIXME_DiagnosticId = "S1134";
+        protected const string FIXME_MessageFormat = "Take the required action to fix the issue indicated by this '" + FIXME_Keyword + "' comment.";
+
+        protected abstract DiagnosticDescriptor TODO_Diagnostic { get; }
```

Comment:
I'd just name it `{Todo,Fixme}Diagnostic` (there's no reason to use underscores)